src/app/exam/options/ExamWhiteList.js

```diff
@@ -1,17 +1,29 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import {SubtitleLine} from '../../commons/titles/TitleLine';
 import {useTextItems} from '../../usecases/TextItemUseCase';
 import {TextInputField} from '../../commons/TextInputForm/TextInputField';
 import {FixedTextInputField} from '../../commons/TextInputForm/FixedTextInputField';
 import './ExamOptions.scss';
 
-const ExamWhiteList = () => {
-  const {textItems, addTextItem, removeTextItem} = useTextItems();
+const ExamWhiteList = ({whitelist, onWhitelistChange}) => {
+  const { textItems: ipAddresses, addTextItem: addIpAddress, removeTextItem: removeIpAddress } = useTextItems(whitelist);
+
+  useEffect(() => {
+    onWhitelistChange(ipAddresses);
+  }, [ipAddresses]);
+
+  const onAddIpAddress = (ipAddress) => {
+    const ipAddressFormat = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
```

Comment:
try this one `'\b((25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.|$)){4}\b'`
here is the reference [link](https://stackoverflow.com/questions/5284147/validating-ipv4-addresses-with-regexp)

Reply:
Yes, it did work.